app/src/main/java/com/mikhailovskii/weatherandroid/ui/sticker_purchase/StickerPurchaseActivity.kt

```diff
@@ -30,37 +30,34 @@ class StickerPurchaseActivity : AppCompatActivity(), StickerPurchaseContract.Sti
         presenter.getStickerPackByName(name!!)
     }
 
-    @SuppressLint("SetTextI18n")
+    override fun onDestroy() {
+        super.onDestroy()
+        presenter.detachView()
+    }
+
     override fun onStickerPackByNameLoaded(stickerPack: StickerPack) {
         val stickersList = ArrayList<String>()
 
-        for (sticker in stickerPack.stickers!!) {
-            stickersList.add(sticker)
+        if (stickerPack.stickers != null && stickerPack.stickers?.isNotEmpty() == true) {
+            for (sticker in stickerPack.stickers!!) {
```

Comment:
@SergeiMikhailovskii you check nullability of `sticekrPack.stickers` twice and still have to use double bang operator. Don't you think you're doing something wrong here? Moreover all you do here is just copying all the data from one list to another. You may try the following
```kotlin
stickerPack.stickers?.apply { stickerList.addAll(this) }
```
or 
```kotlin
stickerPack.stickers?.run { stickerList.addAll(this) }
```
Both will give you the same result.